sdk/python/test_data/pipelines/pipeline_with_parallelfor_parallelism.py

```diff
@@ -24,6 +27,57 @@ def print_text(msg: str):
     print(msg)
 
 
+@component
+def print_int(x: int):
+    print(x)
+
+
+@component
+def list_dict_maker_0() -> List[Dict[str, int]]:
```

Comment:
Can we also add a tests of invalid cases to `compiler_test.py`?

I'm thinking of:
1) An invalid case where we do raise, since type information is not available.
2) A technically invalid case where we do not raise.